### DISCUSSION TITLE
The actual size of the SVG

### DISCUSSION THREAD

**Discussion by Dimon-II** [AngusJohnson/Image32#17] [General]
I'm developing a font editor for pasting colored SVGs with offset and size adjustments.
Using another SVG rendering library, I encountered an error in determining the real size and image offset on some files (with masks and cut-off area) - instead of ~3 digits I get ~7 digits...
I need not viewBox and width / height, but the corners of a real image (paths, rectangles, bitmaps).
Does your library have the ability to get the minimum-maximum coordinates of the entire SVG image or individual named elements?

[Current version with font samples](https://drive.google.com/file/d/1p5RqYvCL38I587jNVuwRHTG5kutkVe-T/view?usp=sharing)

**Comment by AngusJohnson** [comment]
> Does your library have the ability to get the minimum-maximum coordinates of the entire SVG image or individual named elements?

The short answer is no. 

Given that the Viewbox isn't helpful to you, it seems you're wanting the dimensions (bounds) of each individual display element. And that's not readily accessible because each element has different attributes that define or at least affect their bounds (eg 'd' attribute in path elements, 'r' attribute in circle elements, and of course all the complications of assessing dimensions of text elements with seemingly endless font attributes); and don't forget to consider the complications of stroke widths, and various affine transformations that may be applied to individual elements or groups of elements.

**Reply by Dimon-II** [reply]
![image](https://user-images.githubusercontent.com/29166758/218742913-05e1aa02-402f-4f5a-8b40-2013ce731042.png)
No text size needed for the current task - SVGs used in a font cannot have a text tag. 
The real problem is an non-rectangular clip-patch overlaid on another intricately shaped shape.

**Comment by AngusJohnson** [comment]
Yep, I could have added clip paths to my list above. 
What you're wanting is _really_ complicated. But you have my source code so you're welcome to customize it if you wish 😁.
